src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Action } from 'vs/base/common/actions';
+import { localize } from 'vs/nls';
+import { INotebookService, INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import { NotebookComponent } from 'sql/workbench/contrib/notebook/browser/notebook.component';
+import { ICellModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { CellType } from 'sql/workbench/services/notebook/common/contracts';
+import { ToggleableAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
+
+export class CellToolbarAction extends Action {
+
+	constructor(
```

Comment:
I used this pattern so I could pass in id, label, cssClass and tooltip details; however, I do not know what to do with _cellModel and _notebookService though they are necessary for my actions to function.

Reply:
Not sure what you're saying here - they aren't being used so how are they necessary?

Reply:
I think CellToolbarAction is confusing things a bit here (especially when you have other actions like AddCellAction below). Let's get rid of this and define each action on its own first, and then worry about making it more generic (if it makes sense) after the actions are implemented. Does that make sense?

Reply:
Makes sense.